lib/presentation/pages/login_page/login_page.dart

```diff
@@ -20,61 +21,69 @@ class LoginPage extends StatefulWidget {
-                const WelcomeText(),
-                const SizedBox(height: AppDimensions.d40),
-                const WelcomeSubtitle(),
-                const SizedBox(height: AppDimensions.d80),
-                Text(
-                  context.tr.login,
-                  style: context.tht.displayMedium,
-                ),
-                const SizedBox(height: AppDimensions.d20),
-                CustomTextField(
-                  hintText: context.tr.email,
-                  controller: emailController,
-                  validator: (String? emailController) {
-                    return (emailController == null && emailController == '') ? context.tr.registration_page_fieldCanNotBeEmpty : null;
-                  },
-                ),
-                CustomTextField(
-                  hintText: context.tr.password,
-                  controller: passwordController,
-                  hasPassword: true,
-                  validator: (String? passwordController) {
-                    return (passwordController == null && passwordController == '')
-                        ? context.tr.registration_page_fieldCanNotBeEmpty
-                        : null;
-                  },
-                ),
-                const SizedBox(height: AppDimensions.d120),
-                CustomFloatingActionButton(
-                  text: context.tr.login,
-                  onPressed: () {},
-                ),
-                Padding(
-                  padding: const EdgeInsets.only(top: AppDimensions.d20),
-                  child: Text(
-                    context.tr.or,
-                    style: context.tht.headlineSmall,
+            child: Form(
+              key: _formKey,
+              child: Column(
+                children: [
+                  const SizedBox(height: AppDimensions.d40),
+                  const WelcomeText(),
+                  const SizedBox(height: AppDimensions.d40),
+                  const WelcomeSubtitle(),
+                  const SizedBox(height: AppDimensions.d80),
+                  Text(
+                    context.tr.login,
+                    style: context.tht.displayMedium,
                   ),
-                ),
-                TextButton(
-                  onPressed: () => context.router.push(RegistrationRoute()),
-                  child: Text(
-                    context.tr.registration,
+                  const SizedBox(height: AppDimensions.d20),
+                  CustomTextField(
+                    hintText: context.tr.email,
+                    controller: emailController,
+                    validator: (String? value) {
+                      if (value == null || value.isEmpty) {
+                        return context.tr.login_page_fieldCanNotBeEmpty;
+                      }
+                      if (!value.isValidEmail) {
+                        return context.tr.login_page_yourEmailIsIncorrect;
+                      }
+                      return null;
```

Comment:
you use almost the same pattern in the registration so maybe you can make this validation a global function. I would create a Validator class which have these functions for validation, the translations can be `generic_fieldCanNotBeEmpty`